backend/lib/models/NotifyPreference.js

```diff
@@ -0,0 +1,35 @@
+const { Schema, model } = require("mongoose");
+
+const notifyPreferenceSchema = new Schema({
+  message: {
+    instant: { default: true, type: Boolean },
+    daily: { default: false, type: Boolean },
+    weekly: { default: false, type: Boolean },
+    biweekly: { default: false, type: Boolean },
+  },
+  like: {
+    instant: { default: false, type: Boolean },
+    daily: { default: false, type: Boolean },
+    weekly: { default: true, type: Boolean },
+    biweekly: { default: false, type: Boolean },
+  },
+  comment: {
+    instant: { default: false, type: Boolean },
+    daily: { default: true, type: Boolean },
+    weekly: { default: false, type: Boolean },
+    biweekly: { default: false, type: Boolean },
+  },
+  post: {
+    instant: { default: false, type: Boolean },
+    daily: { default: true, type: Boolean },
+    weekly: { default: false, type: Boolean },
+    biweekly: { default: false, type: Boolean },
+  },
+});
+
+const NotifyPreference = model("NotifyPreference", notifyPreferenceSchema);
+
+module.exports = {
+  model: NotifyPreference,
```

Comment:
You don't need to make it a model, just a schema. as it will be part of the `User` model

Reply:
Got it, thanks Zayd